lib/tlsca/ca.go

```diff
@@ -110,37 +128,82 @@ func (id *Identity) Subject() (pkix.Name, error) {
 	subject.Organization = append([]string{}, id.Groups...)
 	subject.OrganizationalUnit = append([]string{}, id.Usage...)
 	subject.Locality = append([]string{}, id.Principals...)
+
+	// DELETE IN (4.4.0)
+	// Groups are marshaled to both ASN1 extension
+	// and old Province section, for backwards-compatibility,
+	// however begin migration to ASN1 extensions in the future
+	// for this and other properties
```

Comment:
If we're going to delete this on a minor version change, we should definitely add checks in `tsh` to warn about outdated cert format when loading identity files.

Maybe we should start explicitly versioning the teleport certificate format and producing readable errors on version incompatibility.  Certificate incompatibility has the potential to fail silently in very confusing ways.

Reply:
We can delete this in major, overall it should not have any impact if we just skip one version, because the new cert fields are not used on the tsh client,  are only in memory in proxy and are not stored anywhere, so it will be safe to delete with no backwards compatibility problems.

Reply:
Oops, this comment was more directed at the *parsing* side of things, see below.